src/HandleMgr.cpp

```diff
@@ -69,26 +69,53 @@ namespace splash
         return full_msg.str();
     }
 
+    void HandleMgr::setFileNameScheme(FileNameScheme fileNameScheme) throw (DCException)
+    {
+        if (this->fileNameScheme == fileNameScheme)
+            return;
+        if (!filename.empty())
+            throw DCException(getExceptionString("setFileNameScheme",
+                                    "Tried to change scheme while file(s) were still open", ""));
+        this->fileNameScheme = fileNameScheme;
+    }
+
     void HandleMgr::open(Dimensions mpiSize, const std::string baseFilename,
             hid_t fileAccProperties, unsigned flags)
+    throw (DCException)
     {
-        this->numHandles = mpiSize.getScalarSize();
         this->mpiSize.set(mpiSize);
         this->filename = baseFilename;
         this->fileAccProperties = fileAccProperties;
         this->fileFlags = flags;
-        this->singleFile = false;
+        // Validation: For parallel files we normally append MPI rank or iteration number.
+        //             This is disabled by using FNS_FULLNAME
+        //             or when the filename already contains an h5-extension.
+        if (fileNameScheme != FNS_FULLNAME && baseFilename.find(".h5") == baseFilename.length() - 3)
+        {
+            if (mpiSize.getScalarSize() > 1)
+            {
+                throw DCException(getExceptionString("open",
+                        "Passed full filename for parallel file operations",
```

Comment:
Then one should pass FNS_FULLNAME to the ctor. If one does otherwise one must understand the implications. This gets more and more complicated and confusing. I'd rather leave this as it is than put even more conditions in here.